components/board/genericlinux/data.go

```diff
@@ -187,35 +185,28 @@ func getPwmChipDefs(pinDefs []PinDefinition) (map[string]pwmChipData, error) {
 
 func getBoardMapping(pinDefs []PinDefinition, gpioChipsInfo map[int]string,
 	pwmChipsInfo map[string]pwmChipData,
-) (map[int]GPIOBoardMapping, error) {
-	data := make(map[int]GPIOBoardMapping, len(pinDefs))
+) (map[string]GPIOBoardMapping, error) {
+	data := make(map[string]GPIOBoardMapping, len(pinDefs))
 
 	// For "use" on pins that don't have hardware PWMs
 	dummyPwmInfo := pwmChipData{Dir: "", Npwm: -1}
 
 	for _, pinDef := range pinDefs {
-		key := pinDef.PinNumberBoard
-
-		var ngpio int
-		for n := range pinDef.GPIOChipRelativeIDs {
-			ngpio = n
-			break // each gpio pin should only be associated with one gpiochip in the config
-		}
-
-		gpioChipDir, ok := gpioChipsInfo[ngpio]
+		key := pinDef.Name
```

Comment:
Mild preference to remove `key` and just use `pinDef.Name` in those places. `key` isn't nearly as descriptive a variable name.